src/GameplaySystem.cpp

```diff
@@ -718,32 +750,40 @@ void GameplaySystem::handle_vehicle_collision(const Event& e) {
     int b_id = e.get_value<int>("b_id", true).first;
     std::cout << a_id << " collided with " << b_id << std::endl;
 
+    bool a_invincible = powerup_datas_[a_id].invincibility > 0.f;
+    bool b_invincible = powerup_datas_[b_id].invincibility > 0.f;
+    bool it_invincible = powerup_datas_[current_it_id_].invincibility > 0.f;
```

Comment:
I don't think this is necessary, the only players who this function cares about are `a` and `b`. Like I think where it's used below isn't necessary, as long as the logic for `a` and `b` are correct.